docs: update ingress monitoring documentation to use ingress-nginx instead of Ambassador (#7548)
prometheus-operator/prometheus-operator: Documentation/user-guides/monitoring-kubernetes-ingress.md

* Deploy the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md)
* Configure monitoring for the ingress-nginx controller
* Verify metrics from the ingress-nginx controller are being scraped correctly

## Prerequisites
* A Kubernetes Cluster
* The [Kubernetes command line tool](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
* The [Helm Kubernetes package manager](https://helm.sh/docs/intro/install/)
* ingress-nginx installed in your cluster

## Deploy kube-prometheus-stack

> The kube-prometheus-stack includes Prometheus Operator.

To deploy the kube-prometheus-stack:

1. Add the prometheus-community Helm repository:

```sh
helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
helm repo update
```

2. Install the kube-prometheus-stack:

```sh
helm install prometheus prometheus-community/kube-prometheus-stack
```

> By default, the prometheus-operator will be installed in the default namespace.
> You may need to update the spec.serviceMonitorSelector.matchLabels in order to correctly monitor ingress-nginx.

## Configure ingress-nginx Monitoring

> The ingress-nginx Service does not expose the Prometheus monitoring port by default. We need to first ensure the Service exposes the port for Prometheus metrics and then create a ServiceMonitor to tell Prometheus to scrape those metrics.

Reviewer: Hi @thaniri, a nit:

```suggestion
After deploying Prometheus, we need to expose ingress-nginx Service to scrape metrics, since it doesn't create that Service by default. Afterwards, we need to create a ServiceMonitor to tell Prometheus to scrape those metrics.
```

Author: Thank you for the review, @heliapb 🎉 

I do not agree with your suggested change, as in this guide I patched the default Service provided by the `ingress-nginx` chart as opposed to creating a new one. Unless there is a prometheus-operator best practice that I am not aware of which recommends creating a new Service for monitoring purposes.

The key difference between my sentence and yours is found here:

```
The ingress-nginx **Service does not expose the Prometheus monitoring port by default**. We need to first **ensure the Service exposes the port** ...
```

Implying a Service _does_ exist, but doesn't expose the port we want.

Whereas yours:

```
After deploying Prometheus, we need to expose ingress-nginx Service to scrape metrics, **since it doesn't create that Service** ...
```

Implies we must create a new Service.

As we don't need to create a new Service, mine reads more correctly. Though, yours would still work! (With a minor change to the guide as well to not include the Service patch.)
